### PR TITLE
Added parameter to skip invalid packages on update

### LINKED ISSUE #412: Skip invalid rpms
`RpmMetadata.Append` should skip invalid rpm files on update: log the error, skip invalid file, continue the update.

### PR DESCRIPTION
**Pull request by olenagerasimova**
Closes #412 
Added parameter to skip invalid packages on update, class to handle exceptions and made `InvalidPackageException` extend `ArtipieException`.

### CODE REVIEW

File: src/main/java/com/artipie/rpm/meta/MergedXml.java [hunk 1]
```diff
@@ -68,4 +69,61 @@ public long count() {
             return this.ids;
         }
     }
+
+    /**
+     * Handles invalid rpm packages.
+     * @param <E> Exception type
+     * @since 1.7
+     */
+    final class InvalidPackage<E extends Throwable> {
```
Comment (genryxy) [inline src/main/java/com/artipie/rpm/meta/MergedXml.java:78]: @olenagerasimova I'd suggest to remove generic type here

**Review by genryxy — CHANGES_REQUESTED** [review]
@olenagerasimova thanks, one suggestion

**Comment by olenagerasimova** [comment]
@genryxy thanks, done

**Review by genryxy — APPROVED** [review]
@olenagerasimova thanks, looks good to me

**Pull request merged**